x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/index.tsx

```diff
@@ -138,16 +142,29 @@ export const AllRules = React.memo<AllRulesProps>(
         pagination,
         rules,
         selectedRuleIds,
+        lastUpdated,
+        showIdleModal,
+        isRefreshPaused,
+        intervalValue,
       },
       dispatch,
-    ] = useReducer(allRulesReducer(tableRef), initialState);
+    ] = useReducer(allRulesReducer(tableRef), { ...initialState, lastUpdated: Date.now() });
     const { loading: isLoadingRulesStatuses, rulesStatuses } = useRulesStatuses(rules);
     const history = useHistory();
     const [, dispatchToaster] = useStateToaster();
     const mlCapabilities = useMlCapabilities();
     const [allRulesTab, setAllRulesTab] = useState(AllRulesTabs.rules);
     const { formatUrl } = useFormatUrl(SecurityPageName.detections);
 
+    // Auto rules info refresh refs
+    const idleTimeoutId = useRef<NodeJS.Timeout | null>(null);
```

Comment:
Such a weird world between NodeJS and the browser. The browser puts all the globals on `window` and then NodeJS puts them all on `global`. But we run the unit tests on the backend within NodeJS and not really the front end.

What's interesting is actually a 3rd way called `globalThis`:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#type-checking-for-globalthis

I don't know if/when that will catch on enough, but in the meantime you can remove that `NodeJS.Timeout` with a handy trick like this:

```ts
const idleTimeoutId = useRef<ReturnType<typeof setTimeout> | null>(null);
```

It's interesting to change that `ReturnType` to be `window.setTimeout` instead but notice that TypeScript didn't really replace it all the way correctly, so it's a bit of workaround to avoid just having the words `NodeJS` in your front end types.